Library/Homebrew/utils/analytics.sh

```diff
@@ -0,0 +1,71 @@
+setup-analytics() {
+  if [[ -z "$HOMEBREW_ANALYTICS" || -n "$HOMEBREW_NO_ANALYTICS" ]]
+  then
+    return
+  fi
```

Comment:
Be aware that the `[[ ... ]] || return` idiom has slightly different behavior: the result of the test becomes the `$?` exit status of the statement in this case, whereas with `if [[ ... ]]` the exit status is still 0 (success). Matters if the exit status of the function will be tested, if `set -e` is enabled, or if it's the last statement run in a script. I'd do an explicit `return 0` in this case to keep exit status in line with the function success.